provider/policy/resource/security.py

```diff
@@ -1,40 +1,399 @@
+                )
+                resources_found.append(
+                    Resource(
+                        digest=resource_digest,
+                        name=data["RoleName"],
+                        details="",
+                        group="",
+                    )
+                )
+                if (
+                    "AssumeRolePolicyDocument" in data
+                    and "Statement" in data["AssumeRolePolicyDocument"]
+                ):
+                    for statement in data["AssumeRolePolicyDocument"]["Statement"]:
+                        resources_found.extend(
+                            self.analyze_assume_statement(resource_digest, statement)
+                        )
+
+        self.resources_found = resources_found
+        return resources_found
+
+    def analyze_assume_statement(
+        self, resource_digest: ResourceDigest, statement
+    ) -> List[Resource]:
+        resources_found = []
+        if "Principal" in statement and "Service" in statement["Principal"]:
+            assuming_services = statement["Principal"]["Service"]
+            if not isinstance(assuming_services, list):
+                assuming_services = [assuming_services]
+            for assuming_service in assuming_services:
+                principal_found: Resource = None
+                if assuming_service in Principals.principals:
+                    principal_found = Resource(
+                        digest=ResourceDigest(
+                            id=assuming_service,
+                            type=Principals.principals[assuming_service]["type"],
+                        ),
+                        name=Principals.principals[assuming_service]["name"],
+                        details="",
+                        group="",
+                    )
+                else:
+                    principal_found = Resource(
+                        digest=ResourceDigest(id=assuming_service, type="aws_general"),
+                        name=assuming_service,
+                        details="",
+                        group="",
+                    )
+                if principal_found is not None:
+                    resources_found.append(principal_found)
+                    self.create_principal_relation(
+                        resource_digest, principal_found.digest
+                    )
+        return resources_found
+
+    def create_principal_relation(self, resource_digest, service_digest):
+        self.relations_found.append(
+            ResourceEdge(
+                from_node=resource_digest, to_node=service_digest, label="assumed by"
+            )
+        )
+
+    @exception
+    def get_relations(self) -> List[ResourceEdge]:
+        additional_relations_found = self.relations_found
+        with ThreadPoolExecutor(15) as executor:
+            results = executor.map(
+                lambda data: self.analyze_role_relations(data), self.resources_found
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Lambda may not be necessary](https://app.codacy.com/gh/Cloud-Architects/cloud-discovery/pullRequest?prid=5655786)